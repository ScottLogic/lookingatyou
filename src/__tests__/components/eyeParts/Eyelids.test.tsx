import { shallow } from 'enzyme';
import React from 'react';
import {
    Eyelids,
    IEyelidsProps,
} from '../../../components/eye/eyeParts/Eyelids';

let props: IEyelidsProps;

describe('Eyelids', () => {
    beforeEach(() => {
        props = {
            transitionStyle: { transition: '' },
            eyeShape: {
                leftX: 0,
                rightX: 0,
                middleY: 0,
                middleX: 0,
                topEyelidY: 0,
                bottomEyelidY: 0,
            },
            cornerShape: {
                leftTop: 0,
                rightTop: 0,
                leftBottom: 0,
                rightBottom: 0,
            },
            scleraRadius: 300,
            bezier: {
                controlOffset: 20,
                scaledXcontrolOffset: 20,
                scaledYcontrolOffset: 30,
            },
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<Eyelids {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });
});
