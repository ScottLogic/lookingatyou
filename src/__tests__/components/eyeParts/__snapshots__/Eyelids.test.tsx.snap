// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eyelids should render correctly 1`] = `
"<svg className=\\"Eyelids\\">
  <path style={{...}} filter=\\"url(#shadowTop)\\" d=\\"M 0 0\\\\n                         A 300 300 0 0 1 0 0\\\\n                         C 0 -30,\\\\n                         20 0, 0 0\\\\n                         C -20 0, 0 -30, 0 0\\" />
  <path style={{...}} filter=\\"url(#shadowBottom)\\" d=\\"M 0 0\\\\n                         A 300 300 0 0 0 0 0\\\\n                         C 0 30,\\\\n                         20 0, 0 0\\\\n                         C -20 0, 0 30, 0 0\\" />
</svg>"
`;
